routes/index.js

```diff
@@ -1041,7 +1041,12 @@ module.exports = function () {
     // Pagination - show 20 items per page:
     const from = req.query.from || 0
     const size = req.query.size || 20
-    const packages = await api.search(`datahub.ownerid="${userProfile.profile.id}"&size=${size}&from=${from}`, token)
+    let packages
+    if (req.query.q) {
+      packages = await api.search(`datahub.ownerid="${userProfile.profile.id}"&q="${req.query.q}"&size=${size}&from=${from}`, token)
+    } else {
+      packages = await api.search(`datahub.ownerid="${userProfile.profile.id}"&size=${size}&from=${from}`, token)
```

Comment:
Why do you need `if/else` at all?  you can just set query parameter to the empty string and it will give you the same exact result (I think)
```
const query = req.query.q || ''
packages = await api.search(`datahub.ownerid="${userProfile.profile.id}"&q="${query}"&size=${size}&from=${from}`, token)
```

Reply:
@zelima it does not work, it returns 0 datasets, we have to replace entire query, so refactored to use your logic